plugins/org.locationtech.udig.project/src/org/locationtech/udig/project/internal/render/impl/RenderMetricsSorter.java

```diff
@@ -97,12 +97,36 @@ private int compareMetrics ( InternalRenderMetrics o1, InternalRenderMetrics o2
         int first = -1;
         int last = 1;
         
-        // check the layer/map blackboard for a preferred or non-preferred renderer 
-        int perferredMapRenderer = rateUsingBlackboardSettings(o1, o1.getRenderContext().getMap().getBlackboard());
-        if (perferredMapRenderer > 0){
+        // check the layer blackboard for a preferred or non-preferred renderer
+        // check the layer blackboards for a preferred or non-preferred renderer
+        int perferredMapRenderer = rateUsingBlackboardSettings(o1, o1.getRenderContext().getLayer()
+                .getBlackboard());
+        if (perferredMapRenderer == 0) {
+        	perferredMapRenderer = rateUsingBlackboardSettings(o1, o1.getRenderContext().getLayer()
+                    .getStyleBlackboard());
+        }
+        int perferredMapRenderer2 = rateUsingBlackboardSettings(o2, o2.getRenderContext().getLayer()
+                .getBlackboard());
+        if (perferredMapRenderer2 == 0) {
+        	perferredMapRenderer2 = rateUsingBlackboardSettings(o2, o2.getRenderContext().getLayer()
+                    .getStyleBlackboard());
+        }       
+        if (perferredMapRenderer > perferredMapRenderer2) {
+            return first;
+        }
+        if (perferredMapRenderer < perferredMapRenderer2) {
+            return last;
+        }
+        
+        // check the map blackboard for a preferred or non-preferred renderer
+        perferredMapRenderer = rateUsingBlackboardSettings(o1, o1.getRenderContext().getMap()
+                .getBlackboard());
+        perferredMapRenderer2 = rateUsingBlackboardSettings(o2, o2.getRenderContext().getMap()
+                .getBlackboard());  
+        if (perferredMapRenderer > perferredMapRenderer2) {
             return first;
         }
-        if (perferredMapRenderer < 0){
+        if (perferredMapRenderer < perferredMapRenderer2) {
```

Comment:
This is the same check like in line 117. I guess it's a different comparision (StyleBlackboard vs. Blackboard) and code would be better readable if the name of the variables have a semantic.
However, looks like duplicated code/checks I don't understand 😬